apps/arweave/src/ar_node_worker.erl

```diff
@@ -1041,18 +1045,17 @@ apply_block(State) ->
 			{noreply, State};
 		{B, [PrevB | _PrevBlocks], {{not_validated, awaiting_nonce_limiter_validation},
 				_Timestamp}} ->
-			case maps:get(nonce_limiter_validation_scheduled, State, false) of
+			H = B#block.indep_hash,
+			case maps:get({nonce_limiter_validation_scheduled, H}, State, false) of
```

Comment:
Is this correct?

Initial chain:
```
A -> B
```
We enter this code path and request validation for B. Let's say another block comes in:
```
A -> B -> C
```
B is still the earliest so we will continue to wait on B's validation to come in (and won't start working on C in the meantime).

Then a bunch more blocks come in:
```
A -> B -> C
    \> D -> E -> F
```
Under the old code even though D was now identified as the earliest not validated in longest chain, we'd continue to wait on B's validation since the `nonce_limiter_validation_scheduled` flag was set. This caused a problem if the VDF server was on a different fork as B would never validate.

With your fix since the `nonce_limit_validation_scheduled` flag is now based on the hash, we enter this code path and see that `{nonce_limiter_validation_scheduled, D}` is not set and so we start validating that. This unblocks the node and it can continue.

B will never validate (and will eventually trigger a `refuse_validation` event which will clear the flag so we stop waiting)

Is that correct?

Reply:
Yes, precisely!